### PR TITLE
Make tests available for Doc-Download-frontend

### PR DESCRIPTION
**Pull request by robbrazil**
When running the tests as part of the ECS build-document-download-frontend-main job we want to use the Dockerfile that the app uses instead of the generic test image. The app uses this Dockerfile when running so we want to be able to spot any issues when the code runs with it.

download FE unit tests to run in the doc download FE docker image, not the generic unit-test-image

This line was removed because nvm is no longer used
**source $(HOME)/.nvm/nvm.sh && nvm install && npm ci --no-audit && npm rebuild node-sass && npm run build**

We also upgraded the base image of the container to bookworm.

The Dockerfile was split in BASE, PRODUCTION, TEST.

The images build and the tests were run, although we need to make sure to pause STAGING AND PRODUCTION pipelines, as they will fail the tests.

This change also caused the notifications-aws to have its document-download-frontend pipeline upgraded to have the test image being in the pipeline

---

🚨⚠️ This will be deployed automatically all the way to production when you click merge ⚠️🚨

For more information, including how to check this deployment on preview or staging first before it goes to production, see our [team wiki section on deployment](https://github.com/alphagov/notifications-manuals/wiki/Merging-and-deploying#deployment)

### CODE REVIEW

**Review by c4seyla — APPROVED** [review]
Looks Good, what is the failing issue?

**Comment by robbrazil** [comment]
Screenshot 2023-12-18 at 16 13 08
Tests are now passing !!!

File: docker/Dockerfile [hunk 1]
```diff
@@ -8,33 +22,89 @@ RUN source ~/.nvm/nvm.sh \
...
+# Make sure the app/ directory is there so that "make bootstrap" can create app/version.py
+RUN mkdir -p app
+
+# Install dev/test requirements
+COPY --chown=notify:notify Makefile requirements.txt requirements_for_test.txt package-lock.json package.json gulpfile.js ./
+RUN make bootstrap
+
+# Copy from the real world, one dir up (project root) into the environment's current working directory
+# Docker will rebuild from here down every time.
+COPY --chown=notify:notify . .
```
Comment (c4seyla) [inline docker/Dockerfile:110]: New line here

Comment (robbrazil) [reply docker/Dockerfile]: done thanks

**Pull request merged**